tests/integration/test_hmm.py

```diff
@@ -0,0 +1,419 @@
+            best = dojo.load_params(params_class=dojo.HMMParams,
+                                    filename=self.temporary_file.name,
+                                    num=num)
+            self.assertEqual(len(best), num)
+
+            for parameters in best:
+                self.assertIsInstance(parameters, dojo.HMMParams)
+
+            self.assertEqual(best[0].__repr__(), best_params)
+
+    def test_score_with_sample_count_and_weights(self):
+        name = "score"
+        turns = 10
+        noise = 0
+        repetitions = 5
+        num_states = 2
+        opponents = [s() for s in axl.demo_strategies]
+        size = 10
+
+        objective = dojo.prepare_objective(name=name,
+                                           turns=turns,
+                                           noise=noise,
+                                           repetitions=repetitions)
+
+        population = dojo.Population(params_class=dojo.HMMParams,
+                                     params_kwargs={"num_states": num_states},
+                                     size=size,
+                                     objective=objective,
+                                     output_filename=self.temporary_file.name,
+                                     opponents=opponents,
+                                     sample_count=2,  # Randomly sample 2 opponents at each step
+                                     weights=[5, 1, 1, 1, 1],
+                                     bottleneck=2,
+                                     mutation_probability = .01,
+                                     processes=1)
+
+        generations = 4
+        axl.seed(0)
+        population.run(generations)
+        self.assertEqual(population.generation, 4)
+
+        # Manually read from tempo file to find best strategy
+        best_score, best_params = 0, None
+        with open(self.temporary_file.name, "r") as f:
+            reader = csv.reader(f)
+            for row in reader:
+                _, mean_score, sd_score, max_score, arg_max = row
+                if float(max_score) > best_score:
+                    best_score = float(max_score)
+                    best_params = arg_max
+
+        # Test the load params function
+        for num in range(1, 4 + 1):
+            best = dojo.load_params(params_class=dojo.HMMParams,
+                                    filename=self.temporary_file.name,
+                                    num=num)
+            self.assertEqual(len(best), num)
+
+            for parameters in best:
+                self.assertIsInstance(parameters, dojo.HMMParams)
+
+            self.assertEqual(best[0].__repr__(), best_params)
+
+    def test_score_with_particular_players(self):
+        """
+        These are players that are known to be difficult to pickle
+        """
+        name = "score"
+        turns = 10
+        noise = 0
+        repetitions = 5
+        num_states = 2
+        opponents = [axl.ThueMorse(),
```

Comment:
+1 for more integration tests but it's not necessary to single out the tricky to pickle strategies. It would be better to choose fast to run strategies such as TitForTat and WinStayLoseShift.

Reply:
I'll change this.